arraycontext/impl/pyopencl/__init__.py

```diff
@@ -190,11 +195,13 @@ def thaw(self, array):
     # }}}
 
     def transform_loopy_program(self, t_unit):
-        try:
-            return self._loopy_transform_cache[t_unit]
-        except KeyError:
-            pass
-        orig_t_unit = t_unit
+        from warnings import warn
+        warn("Using arraycontext.PyOpenCLArrayContext.transform_loopy_program "
+                "to transform a program. This is deprecated and will stop working "
+                "in 2022. Instead, use the PyOpenCLArrayContext specific to your "
+                "package (e.g. meshmode, grudge, etc.), which will know more "
+                "about how to transform your program.",
```

Comment:
```suggestion
                "in 2022. Instead, subclass PyOpenCLArrayContext and implement "
                "the specific logic required to transform the program for your "
                "package or application.",
```
? Or something like that?

Reply:
Done as part of this force-push: https://github.com/inducer/arraycontext/compare/a68d80547bf07de6d209ed756aafbb743dcda2b4..7a8cf8aeb0cf5b0c7f579dea5634d8a3dc70abf0

(added even more words...)